plugins/plugin-java/che-plugin-java-plain/che-plugin-java-plain-server/src/main/java/org/eclipse/che/plugin/java/plain/server/generator/PlainJavaProjectGenerator.java

```diff
@@ -69,12 +82,53 @@ public void onCreateProject(
     String sourceDirWsPath = resolve(projectWsPath, sourceFolders.get(0));
     fsManager.createDir(sourceDirWsPath);
 
-    String mainJavaWsPath = resolve(sourceDirWsPath, FILE_NAME);
-    fsManager.createFile(
-        mainJavaWsPath,
-        getClass().getClassLoader().getResourceAsStream("files/main_class_content"));
+    String mainJavaWsPath = resolve(sourceDirWsPath, MAIN_CLASS);
+    fsManager.createFile(mainJavaWsPath, getResource(MAIN_CLASS));
+
+    createClasspath(projectWsPath, sourceFolders.get(0));
+    createProjectConfig(projectWsPath);
+  }
+
+  private void createProjectConfig(String projectWsPath)
+      throws ConflictException, NotFoundException, ServerException {
+    String projectConfigTemplate = getResource(PROJECT_FILE_RESOURCE);
+    String projectConfContent =
+        resolveVariables(
+            projectConfigTemplate, singletonMap(PROJECT_NAME_PATTERN, projectWsPath.substring(1)));
+
+    String projectConfWsPath = resolve(projectWsPath, ".project");
+    fsManager.createFile(projectConfWsPath, projectConfContent);
+  }
 
-    service.createSimpleProject(prefixURI(projectWsPath), sourceFolders.get(0));
+  private void createClasspath(String projectWsPath, String srcFolder)
```

Comment:
Both createProjectConfig and createClasspath methods have common piece of code. It is better to move to separate method.

Reply:
fixed